### PR TITLE
Bump to v1.5.0 and fix multiple problem instances

### PR DESCRIPTION
**Pull request by jfowkes**
This is the new release PR:

- [x] signpost cache breaking in index.rst and README
- [x] open issue about cache breaking
- [x] bump to new version and date
- [x] update history.rst
- [x] rebuild documentation

We should probably switch to semantic versioning in the next release in keeping with other python packages, this has the advantage that it will allow us to bump the minor version for minor fixes which is probably a good thing.

### CODE REVIEW

**Comment by jfowkes** [comment]
@lindonroberts something like this? Feel free to edit as appropriate.

**Comment by jfowkes** [comment]
@lindonroberts I've hit an issue during testing with re-importing the same problem:
```python
>>> import pycutest
>>> p = pycutest.import_problem('ROSENBR')
>>> p = pycutest.import_problem('ROSENBR')

 ** SUBROUTINE CUTEST_usetup: allocation error for CUTEST_work_global status = 5014
 Execution terminating 
>>>
```
This is because `CUTEST_usetup` is now getting called twice when `CUTEStProblem` is instantiated, this wasn't an issue previously as `CUTEST_usetup` was in the C extension module and a module can only be imported once.

A possible fix (I think) is for `build_interface.py` to store the current `CUTEstProblem` instance and just return it if the user requests the same problem (`problemDir` is the same) and otherwise create a new `CUTEstProblem` instance?

**Comment by lindonroberts** [comment]
@jfowkes - for the allocation error with CUTEST_usetup, could we modify the C interface using the setupCalled variable (set to 1 after CUTEST_[u|c]setup is called, then back to zero after CUTEST_[u|c]terminate is called)? i.e. only call CUTEST_[u|c]setup if setupCalled=0?

**Comment by jfowkes** [comment]
@lindonroberts excellent idea, let me give that a go. I'll also add a test for this behaviour as the CI did not catch it.

**Comment by jfowkes** [comment]
@lindonroberts good idea and I thought this would work, but unfortunately it seems only CUTEST_[u|c]setup returns the bounds `bl,bu` and initial point `x` that we require :disappointed:

**Comment by jfowkes** [comment]
@lindonroberts okay here is a proposed fix using a variation on the singleton design pattern, what do you think?

**Review by lindonroberts — APPROVED** [review]
Nice solution, glad this works! The new tests are useful too

File: pycutest/install_scripts.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
 # Settings
 setup(
     name='PyCUTEst automatic test function interface builder',
-    version='1.3',
+    version='1.5.0',
```
[the remark this answers is not in the document]
Comment (jfowkes) [reply pycutest/install_scripts.py]: Of course, why didn't I think of this?

File: README.rst [hunk 2]
```diff
@@ -2,6 +2,8 @@
 PyCUTEst: A Python interface to the CUTEst Optimization Test Environment |License| |Build Status| |PyPI Version| |Downloads| |Paper|
 ====================================================================================================================================
 
+**Warning: PyCUTEst v1.5.0 broke existing cached problems, when upgrading to v1.5.0 please clear out your cache.**
```
[the remark this answers is not in the document]
Comment (jfowkes) [reply README.rst]: Excellent point, most users are probably not aware of its location

**Pull request merged**